constructor-manager/src/constructor_manager/utils/misc.py

```diff
@@ -0,0 +1,14 @@
+import logging
+from typing import Any, Tuple
+
+logger = logging.getLogger(__name__)
+
+
+def dedup(items: Tuple[Any, ...]) -> Tuple[Any, ...]:
+    """Deduplicate an list of items."""
+    new_items: Tuple[Any, ...] = ()
+    for item in items:
+        if item not in new_items:
+            new_items += (item,)
+
+    return new_items
```

Comment:
```suggestion
    return tuple(dict.fromkeys(items))
```

I'm not sure it's worth having this separate file for just this function (used once from what I see)?

Reply:
Agreed, there were more utilities here, but I ended up refactoring them out